plugin/pkg/scheduler/scheduler.go

```diff
@@ -89,6 +92,12 @@ func (s *Scheduler) Run() {
 
 func (s *Scheduler) scheduleOne() {
 	pod := s.config.NextPod()
+
+	// If the assumed pods store contains this pod there is a conflict,
+	// since the store can't contain a pod without a host, and NextPod
+	// only gives us pods without hosts.
+	s.config.Modeler.ForgetAssumedPod(pod)
```

Comment:
To clarify -- the "this" I was referring to in the first paragraph above refers to keep the code as-is. Relying on a pod being in "assumed" when it shows up again to be scheduled, as the signal to not try again on that machine, seems like not the right mechanism.